dask/dataframe/core.py

```diff
@@ -1520,12 +1511,16 @@ def elemwise(op, *args, **kwargs):
     dsk = dict(((_name, i), (op2,) + frs) for i, frs in enumerate(zip(*keys)))
     dsk = merge(dsk, *[d.dask for d in dasks])
 
-    if columns is not None:
-        return DataFrame(dsk, _name, columns, divisions)
-    else:
-        column_name = name or consistent_name(n for df in dfs
-                                              for n in df.columns)
-        return Series(dsk, _name, column_name, divisions)
+    if columns == no_default:
+        if len(dfs) >= 2 and len(dasks) != len(dfs):
+            # should not occur in current funcs
+            msg = 'elemwise with 2 or more DataFrames and Scalar is not supported'
+            raise NotImplementedError(msg)
```

Comment:
The limitation is caused by if ops requires 2 or more ``dask.DataFrame/Series`` and ``dask.core.Scalar``. Currently there is no operation affected by the limitation.

Explanation is:

- If arguments only contain one ``dask.DataFrame/dask.Series`` (and others are all scalars), metadata will not be changed. Thus, metadata can be predictable.
- If arguments only contain ``dask.DataFrame/dask.Series`` and no ``dask.core.Scalar``, we can apply op to empty-dummies. Thus, metadata can be predictable.
- If arguments contain 2 or more ``dask.DataFrame/dask.Series`` and 1 or more ``dask.core.Scalar``, we cannot perform empty op without knowing ``Scalar`` dtype. This case is currently not supported.